src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,645 @@
+            }
+
+            private bool TryGetVariableName(SyntaxNode node, out ExpressionSyntax identifier)
+            {
+                switch (node.Kind())
+                {
+                    case SyntaxKind.EqualsValueClause:
+                        identifier = SyntaxFactory.IdentifierName(((VariableDeclaratorSyntax)node.Parent).Identifier);
+                        return true;
+                    case SyntaxKind.MultiplyAssignmentExpression:
+                    case SyntaxKind.SimpleAssignmentExpression:
+                        identifier = ((AssignmentExpressionSyntax)node).Left;
+                        return true;
+                }
+
+                identifier = null;
+                return false;
+            }
+
+            private bool TryReplaceWithLocalFunction(QueryExpressionSyntax source, out DocumentUpdate documentUpdate)
+            {
+                if (TryFindParentAssignment(source, out var parentStatement))
+                {
+                    string localFunctionName = GetFreeSymbolName("localFunction", source.GetLocation().SourceSpan.Start);
+                    StatementSyntax internalNodeMethod(ExpressionSyntax expression) => SyntaxFactory.YieldStatement(SyntaxKind.YieldReturnStatement, expression);
+                    if (TryConvertQueryExpression(source, internalNodeMethod, out var qe))
+                    {
+                        if (TryGetTypeSyntax(source, out var typeSyntax))
+                        {
+                            var localFunctionDeclaration = SyntaxFactory.LocalFunctionStatement(
+                                modifiers: default,
+                                returnType: typeSyntax.WithAdditionalAnnotations(Simplifier.SpecialTypeAnnotation),
+                                identifier: SyntaxFactory.Identifier(localFunctionName),
+                                typeParameterList: null,
+                                parameterList: SyntaxFactory.ParameterList(),
+                                constraintClauses: default,
+                                body: SyntaxFactory.Block(
+                                    SyntaxFactory.Token(
+                                        SyntaxFactory.TriviaList(),
+                                        SyntaxKind.OpenBraceToken,
+                                        SyntaxFactory.TriviaList(SyntaxFactory.EndOfLine(Environment.NewLine))),
+                                    SyntaxFactory.SingletonList(qe),
+                                    SyntaxFactory.Token(SyntaxKind.CloseBraceToken)),
+                                expressionBody: null);
+
+                            var localFunctionInvocation = SyntaxFactory.InvocationExpression(SyntaxFactory.IdentifierName(localFunctionName));
+                            SyntaxNode newParentExpressionStatement = parentStatement.ReplaceNode(source, localFunctionInvocation);
```

Comment:
var

Reply:
the string type is not evident here. prefer to define it explicitly.

---
In reply to: [175267759](https://github.com/dotnet/roslyn/pull/25362#discussion_r175267759) [](ancestors = 175267759)